### PR TITLE
Fixes #31211 - catch exceptions from auth modules and treat them as u…

### LINKED ISSUE #31211: Oauth2 token request behaves incorrectly with invalid token
Request to OCS api with incorrect oauth2-Token:

Authorization: Bearer
GmTINDYkByLjFIAQoxHf7KFXr8F3XsPhwApSADkQFRpotfwlnYKeMkj3RHaUNMcZ
GET http://localhost:8001/ocs/v2.php/apps/notifications/api/v1/notifications?format=json

:x: Returns 200 with empty body instead of 401 if token is incorrect 

Request ajax api with incorrect oauth2-Token:
Authorization: Bearer
GmTINDYkByLjFIAQoxHf7KFXr8F3XsPhwApSADkQFRpotfwlnYKeMkj3RHaUNMcZ
GET http://localhost:8001/index.php/apps/files/ajax/getstoragestats.php?dir=%2F

:x:  Returns 500 if token is incorrect

Stacktrace:

```
[Thu Apr 19 16:46:21 2018] Exception: {"Exception":"Exception","Message":"Invalid token","Code":0,"Trace":"#0 \/home\/ilja\/octestinstances\/owncloud\/lib\/private\/User\/Session.php(830): OCA\\OAuth2\\AuthModule->auth(Object(OC\\AppFramework\\Http\\Request))\n#1 \/home\/ilja\/octestinstances\/owncloud\/lib\/base.php(966): OC\\User\\Session->tryAuthModuleLogin(Object(OC\\AppFramework\\Http\\Request))\n#2 \/home\/ilja\/octestinstances\/owncloud\/lib\/private\/legacy\/json.php(70): OC::handleLogin(Object(OC\\AppFramework\\Http\\Request))\n#3 \/home\/ilja\/octestinstances\/owncloud\/lib\/public\/JSON.php(67): OC_JSON::checkLoggedIn()\n#4 \/home\/ilja\/octestinstances\/owncloud\/apps\/files\/ajax\/getstoragestats.php(31): OCP\\JSON::checkLoggedIn()\n#5 \/home\/ilja\/octestinstances\/owncloud\/lib\/private\/Route\/Route.php(155): require_once('\/home\/ilja\/octe...')\n#6 [internal function]: OC\\Route\\Route->OC\\Route\\{closure}(NULL)\n#7 \/home\/ilja\/octestinstances\/owncloud\/lib\/private\/Route\/Router.php(342): call_user_func(Object(Closure), Array)\n#8 \/home\/ilja\/octestinstances\/owncloud\/lib\/base.php(912): OC\\Route\\Router->match('\/apps\/files\/aja...')\n#9 \/home\/ilja\/octestinstances\/owncloud\/index.php(55): OC::handleRequest()\n#10 {main}","File":"\/home\/ilja\/octestinstances\/owncloud\/apps\/oauth2\/lib\/AuthModule.php","Line":53}
```

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…nauthocized

## Description

# TODO - write acceptance tests

## Related Issue
Fixes #31211 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -828,12 +828,16 @@ public function tryTokenLogin(IRequest $request) {
...
-				return $this->loginUser($user, $password);
+			try {
+				$user = $authModule->auth($request);
+				if ($user !== null) {
+					$uid = $user->getUID();
+					$password = $authModule->getUserPassword($request);
+					$this->createSessionToken($request, $uid, $uid, $password);
+					return $this->loginUser($user, $password);
+				}
+			} catch (Exception $ex) {
```
Comment (IljaN) [inline lib/private/User/Session.php]: Add backslash: \Exception

**Comment by IljaN** [comment]
@DeepDiver1975 Shouldn`t there also be a login-failed message appearing for failed tokens to inform Fail2Ban?

**Comment by DeepDiver1975** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request closed without merging**